src/ToolBox/SOS/Strike/strike.cpp

```diff
@@ -4098,6 +4098,164 @@ class DumpHeapImpl
 #endif
 };
 
+/**********************************************************************\
+* Routine Description:                                                 *
+*                                                                      *
+*    This function dumps async state machines on GC heap,          *
+*    displaying details about each async operation found.              *
+*    (May not work if GC is in progress.)                              *
+*                                                                      *
+\**********************************************************************/
+DECLARE_API(DumpAsync)
+{
+    INIT_API();
+    MINIDUMP_NOT_SUPPORTED();
+    if (!g_snapshot.Build())
+    {
+        ExtOut("Unable to build snapshot of the garbage collector state\n");
+        return E_FAIL;
+    }
+
+    WCHAR* type = NULL;
```

Comment:
A nit - you could use ArrayHolder<WCHAR> here to make the cleanup automatic instead of having to delete it in both the try and catch blocks.

Reply:
Fixed